src/bin/pgcopydb/ld_stream.c

```diff
@@ -318,6 +317,29 @@ stream_init_context(StreamSpecs *specs)
 
 	privateContext->lastWriteTime = 0;
 
+	/*
+	 * When streaming is resumed, transactions are sent in full even if we wrote
+	 * and flushed a transactions partially in previous command. This implies
+	 * that, if the last message is B/I/U/D/T, the streaming resumes from the
+	 * same transaction and there's a need to skip some messages.
+	 *
+	 * However, note that if the last message is COMMIT, the streaming will
+	 * resume from the next transaction.
+	 */
+	if (specs->startposComputedFromJSON &&
+		(specs->startposActionFromJSON == STREAM_ACTION_BEGIN ||
+		 specs->startposActionFromJSON == STREAM_ACTION_INSERT ||
+		 specs->startposActionFromJSON == STREAM_ACTION_UPDATE ||
+		 specs->startposActionFromJSON == STREAM_ACTION_DELETE ||
+		 specs->startposActionFromJSON == STREAM_ACTION_TRUNCATE))
+	{
+		privateContext->reachedStartPos = false;
+	}
+	else
+	{
+		privateContext->reachedStartPos = true;
+	}
```

Comment:
It just seems to many lines in `if` statement, is that okay? Should we do direct assignment to variable rather than `if...else`. Second, should I extract a small function for `action` comparision to B/I/U/D/T?

Reply:
That's not necessary.

Reply:
After thinking for a while and making the change, I believe the code is more readable as it is now so maybe we can leave it as it is.